test/e2e/helpers_test.go

```diff
@@ -411,6 +411,9 @@ func deployIngressWithEchoBackends(ctx context.Context, t *testing.T, env enviro
 
 	t.Logf("exposing deployment %s via service", deployment.Name)
 	service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeClusterIP)
+	for i := range service.Spec.Ports {
+		service.Spec.Ports[i].AppProtocol = lo.ToPtr("http")
```

Comment:
Maybe let's leave a comment explaining _why_.
```suggestion
    // Set Service's appProtocol to http so that Kuma load-balances requests on HTTP-level instead of TCP. 
    // TCP load-balancing proved to cause issues with not distributing load evenly in short time spans like in this test.
    // Ref: https://github.com/Kong/kubernetes-ingress-controller/issues/5498
		service.Spec.Ports[i].AppProtocol = lo.ToPtr("http")
```